src/fairchem/core/trainers/base_trainer.py

```diff
@@ -356,17 +361,128 @@ def load_datasets(self) -> None:
+                            )
+                            logging.info(
+                                f"{lr_target} linear references have been saved to: {path}"
+                            )
+
+                distutils.broadcast_object_list(otf_elementrefs, src=0)
+                # make sure all of the element reference modules are on the same device
+                self.elementrefs.update(otf_elementrefs[0])
+                # set config so that references are not refit
+                self.config["dataset"]["transforms"]["element_references"]["fit"][
+                    "fitted"
+                ] = True
+            else:  # load pre-fitted linear references from file
+                self.elementrefs[target] = create_element_references(
+                    file=elementrefs[target].get("file"),
+                )
+
+        # load or fit normalizers for the dataset.
+        normalizers = self.config["dataset"].get("transforms", {}).get("normalizer", {})
         self.normalizers = {}
-        if normalizer:
-            for target in normalizer:
-                self.normalizers[target] = Normalizer(
-                    mean=normalizer[target].get("mean", 0),
-                    std=normalizer[target].get("stdev", 1),
+        for target in normalizers:
+            if target == "fit" and not normalizers["fit"].get("fitted", False):
+                otf_normalizers = [
+                    {target: None for target in normalizers["fit"]["targets"]}
+                ]
+                # only carry out the fit on master and then broadcast
+                if distutils.is_master():
+                    otf_normalizers = [
+                        fit_normalizers(
+                            targets=normalizers["fit"]["targets"],
+                            element_references=self.elementrefs,
+                            dataset=self.train_dataset,
+                            batch_size=normalizers["fit"].get(
+                                "batch_size", self.config["optim"]["batch_size"]
+                            ),
+                            num_batches=normalizers["fit"].get("num_batches"),
+                            num_workers=self.config["optim"]["num_workers"],
+                            seed=self.config["cmd"]["seed"],
+                        )
+                    ]
+                    # save the normalization for possible subsequent use
+                    if not self.is_debug:
+                        save_checkpoint(
+                            otf_normalizers[0],
+                            self.config["cmd"]["checkpoint_dir"],
+                            "normalizers.pt",
+                        )
+                        logging.info(
+                            f"Normalizers for targets {normalizers['fit']['targets']} have been saved to: {path}"
+                        )
+
+                distutils.broadcast_object_list(otf_normalizers, src=0)
```

Comment:
`distutils.broadcast_object_list(otf_normalizers, src=0, device=self.device)` doesn't adding this argument ensure we are sending them to the respective devices?

Reply:
Not sure I understand the docs for that. They say: "device (torch.device, optional) – If not None, the objects are serialized and converted to tensors which are moved to the device before broadcasting."

In this case they will be serialized and put in the master rank device, but I cant tell if that will ensure they will then be on each of the other ranks' devices once the broadcast is finished.